### PR TITLE
MPTCP Support 

### PR DESCRIPTION
**Pull request by pgagne**
Add recipe for testing MPTCP. This is a more final version of https://github.com/LNST-project/lnst/pull/216, which the changes requested, commits squashed, etc.

### CODE REVIEW

File: lnst/Recipes/ENRT/MPTCPRecipe.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        :param config:
+        :return:
+        """
+        for ep_dev in config.test_wide_devices:
+            ep_dev.netns.mptcp.delete_all()
+
+        del config.test_wide_devices
+
+        #reset rp_filter
+        self.matched.host2.run("sysctl -w net.ipv4.conf.all.rp_filter=1")
```
Comment (olichtne) [inline lnst/Recipes/ENRT/MPTCPRecipe.py:131]: are the routes removed automatically when the ip addresses are deconfigured? if not we should also remove those here

**Review by olichtne — CHANGES_REQUESTED** [review]
Overall looks ok, just two minor questions to answer first.

File: lnst/Recipes/ENRT/MPTCPRecipe.py (hunk 1 above)
Comment (pgagne) [reply lnst/Recipes/ENRT/MPTCPRecipe.py:131]: Yeah I had the same question when I was adding the code to reset `rp_filter`. I checked and yes routes get deleted when you deconfig the address.

Comment (olichtne) [reply lnst/Recipes/ENRT/MPTCPRecipe.py:131]: makes sense, resolving this thread

File: lnst/RecipeCommon/MPTCPManager.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        self._mptcp.limits("set", subflows=n)
+
+    @property
+    def add_addr_accepted(self):
+        nl_msg = self._mptcp.limits("show")[0]
+        return nl_msg.get_attr("MPTCP_PM_ATTR_RCV_ADD_ADDRS")
+
+    @add_addr_accepted.setter
+    def add_addr_accepted(self, n):
+        self._mptcp.limits("set", add_addr_accepted=n)
```
[the remark this answers is not in the document]
Comment (pgagne) [reply lnst/RecipeCommon/MPTCPManager.py:111]: Yeah unfortunately `add_addr_accepted` is actually the name of the MPTCP limit setting itself:
```
	ip mptcp limits set [ subflows NR ] [ add_addr_accepted NR ]
```
```
       ip mptcp limits show   get current MPTCP subflow creation limits
       ip mptcp limits set    change the MPTCP subflow creation limits

       SUBFLOW_NR
              specifies the maximum number of additional subflows
              allowed for each MPTCP connection. Additional subflows can
              be created due to: incoming accepted ADD_ADDR option,
              local subflow endpoints, additional subflows started by
              the peer.

       ADD_ADDR_ACCEPTED_NR
              specifies the maximum number of ADD_ADDR suboptions
              accepted for each MPTCP connection. The MPTCP path manager
              will try to create a new subflow for each accepted
              ADD_ADDR option, respecting the SUBFLOW_NR limit.
```

[the remark this answers is not in the document]
Comment (olichtne) [reply lnst/RecipeCommon/MPTCPManager.py:111]: ok makes sense

**Review by olichtne — APPROVED** [review]

**Comment by olichtne** [comment]
@jtluka do you also want to review?

**Comment by jtluka** [comment]
I reviewed the code and it looks ok. Ack to merge.

**Pull request merged**